### ISSUE TITLE
IpIsLocalChecker may throw

### ISSUE THREAD

**Issue by chriseldredge** [Bobris/Nowin#21]
On badly configured machines (like mine), the call to `Dns.GetHostEntry()` may throw an exception like:

    System.Net.Sockets.SocketException: Could not resolve host 'foo.localnet'

This will prevent Nowin from starting up correctly regardless of what urls have been configured for binding.

**Comment by MeirionHughes** [comment]
I have the same with a (very) minimal build for embedded machine: 

```
Unhandled Exception:
System.Net.Sockets.SocketException: Could not resolve host 'microzed-zynq7'
  at System.Net.Dns.Error_11001 (System.String hostName) [0x00000] in <filename                    unknown>:0
  at System.Net.Dns.GetHostByName (System.String hostName) [0x00000] in <filena                   me unknown>:0
  at System.Net.Dns.GetHostEntry (System.String hostNameOrAddress) [0x00000] in                    <filename unknown>:0
  at Nowin.IpIsLocalChecker..ctor () [0x00000] in <filename unknown>:0
  at Nowin.Server.Start () [0x00000] in <filename unknown>:0
  at EchoServer.Program.Main (System.String[] args) [0x00000] in <filename unkn                   own>:0
```

**Comment by Bobris** [comment]
Ouch I nearly forgot about this one. I saw that you fork it. So do you work on fix on your own?

**Comment by dorvan** [comment]
also this initialization cause random problem about binding allocation. and application become unstable

**Comment by MeirionHughes** [comment]
yeah I just tested this; simply catching the Socket Exception does the trick and the manual endpoint will kick in.

**Comment by MeirionHughes** [comment]
Its not great but it fixed the issue on my embedded system. Not sure if you want to handle without resorting to try..catch...

**Comment by Bobris** [comment]
Looks ok for me. Merged.

**Comment by MeirionHughes** [comment]
Do you know when this can be included in a nuget release/pre-release?

**Comment by Bobris** [comment]
I will do it in next 14 hours.

**Comment by Bobris** [comment]
Published as 0.22.1.0
